document_merge_service/api/unoconv.py

```diff
@@ -10,6 +10,25 @@
 )
 
 
+def get_default_timeout():
+    timeout = 55
+    try:  # pragma: no cover
+        import uwsgi
+
+        harakiri = uwsgi.opt.get("harakiri")
+        if harakiri:
+            try:
+                timeout = max(int(harakiri) - 2, 2)
```

Comment:
I would prefer the constants here to be defined somewhere instead of inline.

As a separate question: why 2 seconds, is this considered enough time for any situation?

Reply:
Well, if harakiri is our typical value, timeout will be 58 seconds which is plenty enough. If we are in development timeout will be 2, which is enough, too, the templates in the unitests are merged much faster than 2 seconds. It is important that timeout is a bit shorter than harakiri, In the worst case, `terminate_then_kill()`, will take 1 second, so there is 1 second left, which is ages in computer times. And we can't use `timeout = 1` because then we aren't shorter than harakiri in the worst case. So max(_, 2) is the lowest possible value (timeout has to be a integer, for some reason).

Which leaves the decision to use `harakiri - 2`, e.g 58 seconds, we could probably go much shorter. My idea was that, currently the timeout was 60 enforced by uwsgi, and I don't want to change that significantly. Let me know if tend to a shorter timeout and what's your reasoning. I can explain 58 seconds and 2 seconds, these are natural minima and maxima. Any other value would be arbitrary. I first wrote `int(harakiri * 0.1)` (6 seconds) and I couldn't justify that.

Reply:
Thanks for laying out the reasoning, I think this solution makes perfect sense then. :+1: